File: examples/modulith-clinical-tool-jpa/src/test/java/io/zenwave360/example/clinicaltool/modules/termsandconditions/AcceptedTermsAndConditionsRepositoryIntegrationTest.java
package io.zenwave360.example.clinicaltool.modules.termsandconditions;

import io.zenwave360.example.clinicaltool.common.BaseRepositoryIntegrationTest;
import io.zenwave360.example.clinicaltool.modules.termsandconditions.domain.*;
import jakarta.persistence.EntityManager;
import java.time.*;
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;

class AcceptedTermsAndConditionsRepositoryIntegrationTest extends BaseRepositoryIntegrationTest {

    @Autowired
    EntityManager entityManager;

    @Autowired
    AcceptedTermsAndConditionsRepository acceptedTermsAndConditionsRepository;

    @Test
    void findAllTest() {
        var results = acceptedTermsAndConditionsRepository.findAll();
        Assertions.assertFalse(results.isEmpty());
    }

    @Test
    void findByIdTest() {
        var id = 1L;
        var acceptedTermsAndConditions =
                acceptedTermsAndConditionsRepository.findById(id).orElseThrow();
        Assertions.assertNotNull(acceptedTermsAndConditions.getId());
        Assertions.assertNotNull(acceptedTermsAndConditions.getVersion());
    }

    @Test
    void saveTest() {
        AcceptedTermsAndConditions acceptedTermsAndConditions = new AcceptedTermsAndConditions();
        acceptedTermsAndConditions.setUserId(0L);
        acceptedTermsAndConditions.setTermsAndConditionsId(0L);
        acceptedTermsAndConditions.setAcceptedDate(Instant.now());

        // Persist aggregate root
        var created = acceptedTermsAndConditionsRepository.save(acceptedTermsAndConditions);

        // reloading to get relationships persisted by id
        entityManager.flush();
        entityManager.refresh(created);
        Assertions.assertNotNull(created.getId());
        Assertions.assertNotNull(created.getVersion());
    }

    @Test
    void updateTest() {
        var id = 1L;
        var acceptedTermsAndConditions =
                acceptedTermsAndConditionsRepository.findById(id).orElseThrow();
        acceptedTermsAndConditions.setUserId(0L);
        acceptedTermsAndConditions.setTermsAndConditionsId(0L);
        acceptedTermsAndConditions.setAcceptedDate(Instant.now());

        acceptedTermsAndConditions = acceptedTermsAndConditionsRepository.save(acceptedTermsAndConditions);
        Assertions.assertEquals(0L, acceptedTermsAndConditions.getUserId());
        Assertions.assertEquals(0L, acceptedTermsAndConditions.getTermsAndConditionsId());
        Assertions.assertEquals(Instant.now(), acceptedTermsAndConditions.getAcceptedDate());
    }

    @Test
    void deleteTest() {
        var id = 1L;
        acceptedTermsAndConditionsRepository.deleteById(id);
        var notFound = acceptedTermsAndConditionsRepository.findById(id);
        Assertions.assertFalse(notFound.isPresent());
    }
}

File: examples/modulith-klinical-tool-jpa/src/test/kotlin/io/zenwave360/example/clinicaltool/modules/termsandconditions/AcceptedTermsAndConditionsRepositoryIntegrationTest.kt
package io.zenwave360.example.clinicaltool.modules.termsandconditions

import io.zenwave360.example.clinicaltool.common.BaseRepositoryIntegrationTest
import io.zenwave360.example.clinicaltool.modules.termsandconditions.domain.*
import jakarta.persistence.EntityManager
import java.time.*
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.data.repository.findByIdOrNull

class AcceptedTermsAndConditionsRepositoryIntegrationTest : BaseRepositoryIntegrationTest() {

    @Autowired lateinit var entityManager: EntityManager

    @Autowired lateinit var acceptedTermsAndConditionsRepository: AcceptedTermsAndConditionsRepository

    @Test
    fun findAllTest() {
        val results = acceptedTermsAndConditionsRepository.findAll()
        Assertions.assertFalse(results.isEmpty())
    }

    @Test
    fun findByIdTest() {
        val id = 1L
        val acceptedTermsAndConditions =
            acceptedTermsAndConditionsRepository.findByIdOrNull(id)
                ?: throw NoSuchElementException(" not found with id: $id")
        Assertions.assertNotNull(acceptedTermsAndConditions.id)
        Assertions.assertNotNull(acceptedTermsAndConditions.version)
    }

    @Test
    fun saveTest() {
        val acceptedTermsAndConditions = AcceptedTermsAndConditions()
        acceptedTermsAndConditions.userId = 1L
        acceptedTermsAndConditions.termsAndConditionsId = 1L
        acceptedTermsAndConditions.acceptedDate = Instant.now()

        // Persist aggregate root
        val created = acceptedTermsAndConditionsRepository.save(acceptedTermsAndConditions)

        // reloading to get relationships persisted by id
        entityManager.flush()
        entityManager.refresh(created)
        Assertions.assertNotNull(created.id)
        Assertions.assertNotNull(created.version)
    }

    @Test
    fun updateTest() {
        val id = 1L
        val acceptedTermsAndConditions =
            acceptedTermsAndConditionsRepository.findByIdOrNull(id)
                ?: throw NoSuchElementException(" not found with id: $id")
        acceptedTermsAndConditions.userId = 1L
        acceptedTermsAndConditions.termsAndConditionsId = 1L
        acceptedTermsAndConditions.acceptedDate = Instant.now()

        val updated = acceptedTermsAndConditionsRepository.save(acceptedTermsAndConditions)
        Assertions.assertEquals(0L, updated.userId)
        Assertions.assertEquals(0L, updated.termsAndConditionsId)
        Assertions.assertEquals(Instant.now(), updated.acceptedDate)
    }

    @Test
    fun deleteTest() {
        val id = 1L
        acceptedTermsAndConditionsRepository.deleteById(id)
        val notFound = acceptedTermsAndConditionsRepository.findByIdOrNull(id)
        Assertions.assertNull(notFound)
    }
}
